lib/version_utils.pm

```diff
@@ -234,68 +291,133 @@ sub is_sle {
     return check_version($query, $version, qr/\d{2}(?:-sp\d)?/);
 }
 
+=head2 is_transactional
+
+Returns true if called on a transactional server
+=cut
 sub is_transactional {
     return 1 if is_caasp;
     return check_var('SYSTEM_ROLE', 'serverro');
 }
 
+=head2 is_sles4migration
+
+Returns true if called in a migration scenario
+=cut
 sub is_sles4migration {
     return get_var('FLAVOR', '') =~ /Migration|migrated/ && check_var('SLE_PRODUCT', 'sles');
 }
 
+=head2 is_sles4sap
+
+Returns true if called in a SAP test
+=cut
 sub is_sles4sap {
     return get_var('FLAVOR', '') =~ /SAP/ || check_var('SLE_PRODUCT', 'sles4sap');
 }
 
+=head2 is_sles4sap_standard
+
+Returns true if called in an SAP standard test
+=cut
 sub is_sles4sap_standard {
     return is_sles4sap && check_var('SLES4SAP_MODE', 'sles');
 }
 
+=head2 is_rt
+
+Returns true if called on a real time system
+=cut
 sub is_rt {
     return check_var('SLE_PRODUCT', 'rt');
 }
 
+=head2 is_hpc
+
+Returns true if called in an HPC test
+=cut
 sub is_hpc {
     return check_var('SLE_PRODUCT', 'hpc');
 }
 
+=head2 is_released
+
+Returns true if called on a released build
+=cut
 sub is_released {
     return get_var('FLAVOR') =~ /Incidents/ || get_var('FLAVOR') =~ /Updates/;
 }
 
+
+=head2 is_staging
+
+Returns true if called in staging
+=cut
 sub is_staging {
     return get_var('STAGING');
 }
 
+=head2 is_storage_ng
+
+Returns true if storage_ng is used
+=cut
 sub is_storage_ng {
     return get_var('STORAGE_NG');
 }
 
+=head2 is_upgrade
+
+Returns true in upgrade scenarios
+=cut
 sub is_upgrade {
     return get_var('UPGRADE') || get_var('ONLINE_MIGRATION') || get_var('ZDUP') || get_var('AUTOUPGRADE') || get_var('LIVE_UPGRADE');
 }
 
+=head2 is_sle12_hdd_in_upgrade
+
+Returns true if called in SLES12 upgrade scenario
+=cut
 sub is_sle12_hdd_in_upgrade {
     return is_upgrade && is_sle('<15', get_var('HDDVERSION'));
 }
 
+=head2 is_desktop_installed
+
+Returns true if a desktop is installed
+=cut
 sub is_desktop_installed {
     return get_var("DESKTOP") !~ /textmode|minimalx/;
 }
 
+=head2 is_system_upgrading
+
+#TODO this should be documented
+=cut
 sub is_system_upgrading {
     # If PATCH=1, make sure patch action is finished
     return is_upgrade && (!get_var('PATCH') || (get_var('PATCH') && get_var('SYSTEM_PATCHED')));
 }
 
+=head2 is_pre15
+
+Returns if system is older than SLE or Leap 15
+=cut
 sub is_pre_15 {
     return (is_sle('<15') || is_leap('<15.0')) && !is_tumbleweed;
 }
 
+=head2 is_aarch64_uefi_boot_hdd
+
+Returns true if system is aarch64 with uefi and shall boot an hdd image
+=cut
 sub is_aarch64_uefi_boot_hdd {
     return get_var('MACHINE') =~ /aarch64/ && get_var('UEFI') && get_var('BOOT_HDD_IMAGE');
 }
 
+=head2 is_server
+
+Returns true if executed on a server pattern
```

Comment:
I think we need to mention also something about SLES for SAP and SLES for migration .

Reply:
Added it